pkg/pretty_errors/pretty_error.go

```diff
@@ -0,0 +1,74 @@
+package pretty_errors
+
+import (
+	"errors"
+)
+
+const (
+	// GenericErrorCode is for errors which does not implement Prettifier and cannot be recognized with a default strategy
+	GenericErrorCode = "GenericError"
+)
+
+// Error describes the error with details, organized in a standard structure.
+// Error also implements `error` interface.
+type Error struct {
+	// Original error, might be empty.
+	Original error `json:"-"`
+	// Message is a custom pretty error message. Might be different from the original error message
+	Message string `json:"message"`
```

Comment:
I would like to suggest to also introduce a suggestion property used to store suggestions to fix the given error.